### ISSUE TITLE
[Tech Request]: In consistent null handling in loading

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#12405]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 1.0-dev 9be612378
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

When loading CSV, other databases allow user to set null string in order to indicate null value.
For example, see https://www.postgresql.org/docs/current/sql-copy.html for postgres's syntax.

In https://github.com/matrixorigin/MO-Cloud/issues/1583, probably customer really mean null = '\\N'.   Notice that this \\N here is not newline.

We need to support this feature.    At the same time, current behavior is in consistent.

```
create table tt (i int, t text, j int, z text);

fengt> load data inline format='csv', data='1, \\N, \\N, \\N' into table tt;
Query OK, 1 row affected
Time: 0.301s
fengt> select * from tt;
+---+-----+--------+-----+
| i | t   | j      | z   |
+---+-----+--------+-----+
| 1 |  \N | <null> |  \N |
+---+-----+--------+-----+
```

\N is treated as null in the int column, but as string \N in text column.   We really should complete support null string and in this case, we should report j as invalid int format.   Like the following, 

```
fengt> fengt> load data inline format='csv', data='1, xxx, xxx, xxx' into table tt;
(1064, 'SQL parser error: You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at line 1 column 5 near "fengt> load data inline format=\'csv\', data=\'1, xxx, xxx, xxx\' into table tt";')
fengt>
```

For the int column, regular invalid int format will be a parse error.

**Comment by ck89119** [comment]
csvReader已经在这个issue https://github.com/matrixorigin/MO-Cloud/issues/1583 中重构，现在支持\N作为null值导入，且不同数据类型的行为是一致的
<img width="958" alt="image" src="https://github.com/matrixorigin/matrixone/assets/4971300/c12a60f1-7da9-4a7c-bb76-5c87365f4920">

**Comment by aressu1985** [comment]
fixed: 275af9cc9a4d113bd5ddcb5381eb9a9d15e0dd87

mysql> create database test;
Query OK, 1 row affected (0.06 sec)

mysql> use test;
Database changed
mysql> create table tt (i int, t text, j int, z text);
Query OK, 0 rows affected (0.04 sec)

mysql> load data inline format='csv', data='1, \\N, \\N, \\N' into table tt;
ERROR 20101 (HY000): internal error: the input value 'N' is not int32 type for column 2
mysql> load data inline format='csv', data='1,\\N,\\N,\\N' into table tt ;
Query OK, 1 row affected (0.02 sec)

mysql> select * from tt;
+------+------+------+------+
| i    | t    | j    | z    |
+------+------+------+------+
|    1 | NULL | NULL | NULL |
+------+------+------+------+
1 row in set (0.00 sec)

mysql>
